### PR TITLE
tests(inventory): Add unit tests for inventory methods

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: pkg/controllers/machineinventory/machineinventory.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (h *handler) planMachineInventoryIndexer(obj *v1beta1.MachineInventory) ([]
 // initializeHandler creates the machine inventory plan secret and sets the initial plan checksum
 func (h *handler) initializeHandler(obj *v1beta1.MachineInventory, status v1beta1.MachineInventoryStatus) ([]runtime.Object, v1beta1.MachineInventoryStatus, error) {
 	if obj == nil || obj.DeletionTimestamp != nil {
-		return nil, obj.Status, generic.ErrSkip
+		return nil, status, generic.ErrSkip
```
Comment (fgiudici) [inline pkg/controllers/machineinventory/machineinventory.go:97]: ouch... good catch!

File: pkg/controllers/machineinventory/machineinventory_test.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	Describe("onInventoryPlanChange", func() {
+		It("Fills the checksums", func() {
+			sec := &corev1.Secret{
+				TypeMeta:   metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{},
+				Type:       "elemental.cattle.io/plan",
+				Data: map[string][]byte{
+					"plan":             []byte("Star Platinum"),
+					"applied-checksum": []byte("Magician's Red"),
+					"failed-checksum":  []byte("Hermit Purple"),
```
Comment (fgiudici) [inline pkg/controllers/machineinventory/machineinventory_test.go:71]: lol

File: .codecov.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      default: false
+      cli:
+        paths:
+          - "cmd/"
+        target: 75%
+        informational: true
+      pkg:
+        paths:
+          - "pkg/"
+        target: 75%
```
Comment (fgiudici) [inline .codecov.yaml:15]: So, here we enforce 75% coverage on pkg to merge, right? 
Aren't we well below 75% right now ? Do we need an "informational: true" here too?

**Review by fgiudici — COMMENTED** [review]
Much more unit tests 🎉  With the bonus of the fix of a potential panic in the code 🙌🏼 
Just a doubt if the codecov config is a bit too strict 🤔

File: .codecov.yaml (hunk 3 above)
Comment (Itxaka) [reply .codecov.yaml:15]: to be honest I was trying to trigger codecov to post in the PR and I didnt knew how. So I copied the codecov config from elemental-cli directly, but it still wont post the results, no idea why. 

So yeah dont take this into account, its meant to be dropped.

codecov just hates me :/

File: pkg/controllers/machineinventory/machineinventory.go (hunk 1 above)
Comment (Itxaka) [reply pkg/controllers/machineinventory/machineinventory.go:97]: yeah took me a while to understand why my simple test was panicking, I thought I borked the fake methods lol

**Review by fgiudici — APPROVED** [review]

File: .codecov.yaml (hunk 3 above)
Comment (Itxaka) [reply .codecov.yaml]: I set it to informational only, just to have the config in there as I cant make codecov work for this PR for some reason :/

**Pull request merged**